src/utils/rand/mod.rs

```diff
@@ -0,0 +1,87 @@
+// Copyright 2019-2025 ChainSafe Systems
+// SPDX-License-Identifier: Apache-2.0, MIT
+
+use rand::{CryptoRng, Rng, RngCore, SeedableRng as _};
+
+/// A wrapper of [`rand::thread_rng`] that can be overridden by reproducible seeded
+/// [`rand_chacha::ChaChaRng`] via `FOREST_TEST_RNG_FIXED_SEED` environment variable.
+/// This is required for reproducible test cases for normally non-deterministic methods.
+pub fn forest_rng() -> impl Rng + CryptoRng {
+    forest_rng_internal(ForestRngMode::ThreadRng)
+}
+
+/// A wrapper of [`rand::rngs::OsRng`] that can be overridden by reproducible seeded
+/// [`rand_chacha::ChaChaRng`] via `FOREST_TEST_RNG_FIXED_SEED` environment variable.
+/// This is required for reproducible test cases for normally non-deterministic methods.
+pub fn forest_os_rng() -> impl Rng + CryptoRng {
+    forest_rng_internal(ForestRngMode::OsRng)
+}
+
+enum ForestRngMode {
+    ThreadRng,
+    OsRng,
+}
+
+fn forest_rng_internal(mode: ForestRngMode) -> impl Rng + CryptoRng {
```

Comment:
Let's add a test for this. If the variable is set, it should produce the same results; otherwise, it should produce different results.

Reply:
Done.